

# ARC Friend

For my ARC DECRA application I wanted a list of citations, including citations, journal impact factor, and AltMetrics score.

This script parses the file generated by the ORCID export function, then uses public APIs to get Altmetric, journal impact factor (TODO), and citations. Unfortunately there is no public journal impact factor API, so the user will have to fill that in manually (for now?).

It kind of works OK, but I have only tested it with my own bibliography. There are likely hundreds of edge cases and weird encodings this script will mangle.

## Dependencies

Only bibtexparser (`pip3 install bibtexparser`). I have not tested this script under Python 2.x.

## Usage:

First, export your ORCID profile using the button at https://orcid.org/my-orcid:

![ORCID button](button.png)

Then, move the exported works.bib into the current directory and run this:

    python parseBibAndMakeTable.py

That will print the final table to standard out.

Alternatively, you can specify the path:

    python parseBibAndMakeTable.py --file /home/potato/Downloads/otherworks.bib

assumes that you renamed your ORCID export to otherworks.bib.

Example output:

```
Citations and Altmetric scores obtained on 22th January, 2019.
Citations obtained from CrossRef.
1. Mousavi-Derazmahalleh, M., Bayer, P.E., Hane, J.K., Babu, V., Nguyen, H.T., Nelson, M.N., Erskine, W., Varshney, R.K., Papa, R., Edwards, D., 2019. Adapting legume crops to climate change using genomic approaches. Plant, Cell and Environment., 42, 6--19.
Impact Factor = FILLME
Citations = 3
Altmetric score = 9.556

2. Taylor, C.M., Kamphuis, L.G., Zhang, W., Garg, G., Berger, J.D., Mousavi-Derazmahalleh, M., Bayer, P.E., Edwards, D., Singh, K.B., Cowling, W.A., Nelson, M.N., 2018. INDEL variation in the regulatory region of the major flowering time gene LanFTc1is associated with vernalization response and flowering time in narrow-leafed lupin (Lupinus angustifoliusL.). Plant, Cell & Environment.
Impact Factor = FILLME
Citations = 4
Altmetric score = 0.25

3. Mousavi-Derazmahalleh, M., Nevado, B., Bayer, P.E., Filatov, D.A., Hane, J.K., Edwards, D., Erskine, W., Nelson, M.N., 2018. The western Mediterranean region provided the founder population of domesticated narrow-leafed lupin. TAG. Theoretical and applied genetics. Theoretische und angewandte Genetik, 131, 2543--2554.
Impact Factor = FILLME
Citations = 0
Altmetric score = 2.35

4. Bayer, P.E., Edwards, D., Batley, J., 2018. Bias in resistance gene prediction due to repeat masking. Nature Plants.
Impact Factor = FILLME
Citations = 1
Altmetric score = 52.3

5. Melonek, J., Zhou, R., Bayer, PE., Edwards, D., Stein, N., Small, I., 2018. High intraspecific diversity of Restorer-of-fertility-like genes in barley The Plant journal : for cell and molecular biology.
Impact Factor = FILLME
Citations = 0
Altmetric score = 6.45
```


This still needs a lot of manual cleaning - I'd bold my name, I need to manually insert the impact factor, there are some additional spaces and missing spaces here and there.... Use at your own peril :)


The script also writes a log file (`arc-friend.log`) for those papers it cannot correctly format:

```
INFO:root:Skipping cited dataset Repbase Repeats And R-Gene Domains Contained Therein, 10.5281/zenodo.1172257 at Zenodo (for now?)
INFO:root:Skipping cited dataset Clover V3 Assembly, 10.5281/ZENODO.1095215 at Zenodo (for now?)
INFO:root:An efficient approach to BAC based assembly of complex genomes has no doi, skipping (for now?)
INFO:root:Skim-Based Genotyping by Sequencing Using a Double Haploid Population to Call SNPs, Infer Gene Conversions, and Improve Genome Assemblies has no doi, skipping (for now?)
INFO:root:High-resolution molecular karyotyping uncovers pairing between ancestrally related Brassica chromosomes. has no doi, skipping (for now?)
INFO:root:Cannot get Altmetric score for 10.17660/ActaHortic.2013.1005.1. Characterising diversity in the Brassica genomes
```

